### PR TITLE
#2006 - Fix error not show up on large image when uploading profile picture

### LINKED ISSUE #2006: Avatar -> blank pop up is displayed if user select 'jpg' file without preview
1. Open FP web app and login using valid email/password
2. navigate to user profile and click on camera icon to upload a avatar
3. select some random image with no preview

Issue: You will observe that the pop up to select the picture is BLANK.
Refer below screenshot and GIF.

ezgif com-video-to-gif-5
Screen Shot 2020-11-30 at 11 00 07 PM

Please note this issue exist in PROD too. Thanks.

### PR DESCRIPTION
**Pull request by linadinh2602**
### What does this pull request aim to achieve? 💯
#2006 - Fix error not show up on large image when uploading profile picture
_Please be concise and link any related issue(s):_ Fix #2006 
After Fix:
image

- [x] I have checked that no one else is working on similar changes.
- [x] I have read the latest [**README**](README.md) and understand the development workflow.
- [x] The name of this branch is: **`feature/<branch_name>`** for a regular feature, **`hotfix/<branch_name>`** for a hotfix, and **`service/<branch_name>`** for a microservice, where `<branch_name>` briefly describes the issue(s) resolved and is prefixed with the issue number(s) (e.g. `1127-1130-update-git-branching-model`). The branch is created in a cloned version of this repo, **not a forked repo**.
- [x] This branch is rebased/merged with the **latest staging** (or the **latest production** if it is a hotfix).
- [x] There are no merge conflicts.
- [x] There are no console warnings and errors.
- [x] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".
- [x] The title describes the issue(s) being addressed using format: **`<issue_numbers> - <brief_issue_description>`** (e.g. `#1127/#1130 - Update Git Branching Model`)

### CODE REVIEW

File: client/src/components/Picture/UploadPic.js [hunk 1]
```diff
@@ -249,31 +249,35 @@ const UploadPic = ({ cameraIconSize, gtmPrefix, user }) => {
...
-                minHeight={250}
-                minWidth={250}
-                circularCrop={true}
-                ruleOfThirds
-                keepSelection={true}
-                onImageLoaded={imageLoaded}
-                onChange={(newCrop) => setCrop(newCrop)}
-              />
-            )}
+                <ReactCrop
```
Comment (linadinh2602) [inline client/src/components/Picture/UploadPic.js:252]: Not sure if my current linter has the right style? What is the guidelines for indentation here?

Comment (ddveloper) [reply client/src/components/Picture/UploadPic.js:252]: it seems my side 235-238, 252-263 and 266-280 has one tab less than yours @linadinh2602  (while I don't know if there is a specification in our repos)

**Review by ddveloper — DISMISSED** [review]
LGTM, thanks @linadinh2602 👍

**Comment by rabbi08** [comment]
Thanks @linadinh2602 
I have following observation while verifying the fix on feature branch
- If avatar already exist and user try to upload new avatar without jpg preview, nothing happens.
- BUT, if user does not have avatar i.e. user profile logo exist, and then user try to upload the jpg without preview, blank error message is still displayed. 

Please see below GIF.
ezgif com-video-to-gif-2

**Comment by linadinh2602** [comment]
@rabbi08 : I have tried reaching all the code paths with both error images and working images. However, I cannot reproduce what you see. Do you know a specific way to reproduce? Or can you share the image that didn't work for you, so that I can try it on my end?

**Comment by rabbi08** [comment]
> @rabbi08 : I have tried reaching all the code paths with both error images and working images. However, I cannot reproduce what you see. Do you know a specific way to reproduce? Or can you share the image that didn't work for you, so that I can try it on my end?

Hi @linadinh2602 
I have onedrive configured on machine, so may be due to network slowness, at times it does not loads the preview of the data. so then this issue comes up. As of now as per my investigation this issue links with the slow network speed, where user just select random jpg without waiting for the preview to load and then upload in the application.
I hope that this defines the issue somewhat. Thanks

**Comment by linadinh2602** [comment]
@rabbi08 : In this case of network slowness that leads to image not showing in the preview pane, then I think what you are seeing is the current expected behavior. Once the image load up, the preview will show with the correct buttons. I'm happy to work to improve this experience, but I think I'll push this PR out first to resolve the other issue that this ticket is addressing, then file a new issue for this network problem. What do you think?

**Comment by rabbi08** [comment]
> @rabbi08 : In this case of network slowness that leads to image not showing in the preview pane, then I think what you are seeing is the current expected behavior. Once the image load up, the preview will show with the correct buttons. I'm happy to work to improve this experience, but I think I'll push this PR out first to resolve the other issue that this ticket is addressing, then file a new issue for this network problem. What do you think?

@linadinh2602 
Actually this issue is for both, the pop up is blank:

1. either if the file size>5MB : Now PASS in this feature build
2. preview of the pic is not uploaded: still FAIL

Pease see below GIF:

ezgif com-video-to-gif-5

So we have fixed #1 in this PR only. great if we can fix the other one here only as else we need another issue and PR. Let me know your thoughts. Thanks.

**Comment by linadinh2602** [comment]
@rabbi08 sorry, I think I am confused from seeing your gif. From what I see, you do two things
1) Try to upload a large image => it fails and display the error message, which is correct
2) Then you try to upload a regular image => the preview show up => you click upload => it uploads. This is also correct.
Then, what is the problem here?

**Comment by rabbi08** [comment]
> @rabbi08 sorry, I think I am confused from seeing your gif. From what I see, you do two things
> 
> 1. Try to upload a large image => it fails and display the error message, which is correct
> 2. Then you try to upload a regular image => the preview show up => you click upload => it uploads. This is also correct.
>    Then, what is the problem here?

in the gif, its only for large size error, you have already fixed. 
Not in this recent GIF, I am now requesting to fix/handle network issue while uploading, also. Let me know if there any queries. 
Thanks @linadinh2602

**Comment by linadinh2602** [comment]
@rabbi08 Seems like the only fix for the network issue is that there is some activity indicator in the view to show the user that the image is loading. I can add that in to help with the visual feedback. Do you agree with this approach?

**Comment by rabbi08** [comment]
@linadinh2602 
Actually the problem is when there is no preview is not available for the selected image. 

But, I am think this is very unique and corner case which can be fixed in later stage. We can merge this issue which fix for 'large file size; images. The fix is working fine when user try to upload image >5MB. 

Thanks

**Review by rabbi08 — APPROVED** [review]

**Review by ddveloper — APPROVED** [review]

**Pull request merged**